src/common/Common.ts

```diff
@@ -131,7 +127,29 @@ export class CommandLineUtils {
         }
     }
 
-    public static createFlag(type: FlagsConfigType, isRequired: boolean): any {
+    /**
+     * Checks to see if a platform flag has a valid value.
+     * @param platformFlag The platform flag.
+     * @param includeDesktop Indicates whether Desktop is allowed as a target platform. Defaults to false.
+     * @returns True if flag is valid.
+     */
+    public static platformFlagIsValid(
+        platformFlag: string,
+        includeDesktop = false
+    ) {
+        return (
+            CommandLineUtils.platformFlagIsIOS(platformFlag) ||
+            CommandLineUtils.platformFlagIsAndroid(platformFlag) ||
+            (includeDesktop &&
+                CommandLineUtils.platformFlagIsDesktop(platformFlag))
+        );
+    }
+
+    public static createFlag(
+        type: FlagsConfigType,
+        isRequired: boolean,
+        supportsDesktop = false
```

Comment:
I don't have great visibility on how plugins are going to use this, but in the absence of that visibility it feels a lot like pushing plugin-specific considerations down into the core library.

What would be a typical use case for one of our consuming plugins, here?

Reply:
In `lwc-dev-mobile` we define a number of plugin commands. Each command has a bunch of flags and there are a few flags (such as ApiLevel, Platform, LogLevel) that are  common to all commands. So the logic around common flags had been placed in `lwc-dev-mobile-core` and I'm just building on that.

As an example, here's what the `flags` definition for an updated `Preview` command would look like:
```
    public static readonly flags = {
        ...CommandLineUtils.createFlag(FlagsConfigType.ApiLevel, false),
        ...CommandLineUtils.createFlag(FlagsConfigType.LogLevel, false),
        ...CommandLineUtils.createFlag(FlagsConfigType.Platform, true, true),
        componentname: Flags.string({
            char: 'n',
            description: messages.getMessage('componentnameFlagDescription'),
            required: true
        }),
        configfile: Flags.string({
            char: 'f',
            description: messages.getMessage('configFileFlagDescription'),
            required: false,
            default: ''
        }),
        projectdir: Flags.string({
            char: 'd',
            description: messages.getMessage('projectDirFlagDescription'),
            required: false,
            default: process.cwd()
        }),
        .
        .
        .
    };
```

Reply:
Ah, interesting. Since there is just a handful or arguments and types associated with this functionality, I'm inclined to go with the change as coded, then. Pedantic overthinking follows below. 😉

The thing that is getting me is that we're effectively bifurcating the input parameter to `createFlag()` needed to configure the same input value: in this case, `type` and `supportsDesktop`. In a world where we needed to scale out this method to any number of supported flags, ideally we'd provide an input parameter data structure where each flag can specify further constraints. E.g.:

```typescript
CommandLineUtils.createFlag(
    {
        type: FlagsConfigType.Platform,
        supportedPlatforms: (PlatformTypeIOS | PlatformTypeAndroid | PlatformTypeDesktop
    },
    true
)
```

Or similar to Node.js methods (like `fs.readFile()`) that provide blanket "options" parameters:

```typescript
CommandLineUtils.createFlag(
    FlagsConfigType.Platform,
    true,
    {
        supportedPlatforms: PlatformTypeIOS | PlatformTypeAndroid | PlatformTypeDesktop
    }
)
```

Anyway, food for thought if we find ourselves getting unwieldy in our input management here.

Reply:
I could also update our enum definition

From
```
export enum FlagsConfigType {
    Platform,
    ApiLevel,
    LogLevel,
    Json
}
```

To
```
export enum FlagsConfigType {
    MobileOnlyPlatforms,
    MobileAndDesktopPlatforms,
    ApiLevel,
    LogLevel,
    Json
}
```

Then we could just call `CommandLineUtils.createFlag(FlagsConfigType.MobileAndDesktopPlatforms, ...)`. But this type of change (or the one you suggested) would require me to touch `ALL` of the commands and update them, whereas the approach I have added here in this PR uses a param with default value so as to not touch many of the commands and only touch the code for the `Preview` command.